src/rviz/InteractiveMarkerViewer.cpp

```diff
@@ -53,6 +53,63 @@ SkeletonMarkerPtr InteractiveMarkerViewer::addSkeletonMarker(
   return marker;
 }
 
+//==============================================================================
+void InteractiveMarkerViewer::updateSkeletonMarkers()
+{
+  // Update SkeletonMarkers from the world.
+  if (mWorld)
+  {
+    for (std::size_t i = 0; i < mWorld->getNumSkeletons(); ++i)
+    {
+      const dart::dynamics::SkeletonPtr skeleton = mWorld->getSkeleton(i);
+
+      if (skeleton)
+      {
+        // Either a new SkeletonMarker or a previously-inserted SkeletonMarker
+        auto result = mSkeletonMarkers.emplace(
+            skeleton,
+            std::make_shared<SkeletonMarker>(
+                nullptr, &mMarkerServer, skeleton, mFrameId));
+
+        std::unique_lock<std::mutex> skeleton_lock(
+            skeleton->getMutex(), std::try_to_lock);
+        if (skeleton_lock.owns_lock())
+          result.first->second->update();
+      }
+    }
+  }
+
+  // Clear removed skeletons
+  auto it = std::begin(mSkeletonMarkers);
+  while (it != std::end(mSkeletonMarkers))
+  {
+    // If there is no underlying world and the skeleton exists,
+    // update the corresponding marker.
+    if (!mWorld && it->first)
+    {
+      std::unique_lock<std::mutex> skeleton_lock(
```

Comment:
As we discussed offline, I think this function might be clearer if we have the insert loop above (without updating), then do all the updating in this loop over `mSkeletonMarkers` since we're iterating over all of them anyway. I think this could also allow us to merge the increments/erases into a single if condition.